File: C/Linked_List/SinglyLL_DISPLAY.c
#include<stdio.h>
#include<stdlib.h>

#pragma pack(1)
struct node 
{
  int data;
  struct node *next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void Display(PNODE first)
{
  while(first != NULL)
  {
    printf("| %d | -> ", first->data);
    first = first->next;
  }
  printf("NULL\n");
}

void InsertFirst(PPNODE first, int iNo)
{
  PNODE newn = NULL;

  newn = (PNODE)malloc(sizeof(NODE));

  newn->data = iNo;
  newn->next = NULL;

  if(*first == NULL)
  {
    *first = newn;
  }
  else
  {
    newn->next = *first;
    *first = newn;
  }
}

void InsertLast(PPNODE first, int iNo)
{
  PNODE newn = NULL;
  PNODE temp = NULL;

  newn = (PNODE)malloc(sizeof(NODE));

  newn->data = iNo;
  newn->next = NULL;

  if(*first == NULL)
  {
    *first = newn;
  }
  else
  {
    temp = *first;

    while(temp->next != NULL)
    {
      temp = temp->next;
    }
    temp->next = newn;
  }
}

int main()
{
  PNODE head = NULL;
  int iRet = 0;

  InsertFirst(&head,51);
  InsertFirst(&head,21);
  InsertFirst(&head,11);

  InsertLast(&head,101);
  InsertLast(&head,111);
  InsertLast(&head,121);

  Display(head);

  return 0;
}

File: Java/Linked_List/SinglyLL_Display.java
package Linked_List;

class node
{
  public int data;
  public node next;

  node(int no)
  {
    this.data = no;
    this.next = null;
  }
}

class SinglyLL 
{
  private node first;
  private int iCount;

  public SinglyLL()
  {
    this.first = null;
    this.iCount = 0;
  }

  public void Display()
  {
    node temp = first;

    while(temp != null)
    {
      System.out.print("| "+temp.data+" | -> ");
      temp = temp.next;
    }
    System.out.println("null");
  }

  public void InsertFirst(int iNo)
  {
    node newn = new node(iNo);

    if(first == null)
    {
      first = newn;
    }
    else
    {
      newn.next = first;
      first = newn;
    }
    iCount++;
  }

  public void InsertLast(int iNo)
  {
    node newn = new node(iNo);

    if(first == null)
    {
      first = newn;
    }
    else
    {
     node temp = first;

     while(temp.next != null)
     {
        temp = temp.next;
     }
     temp.next = newn;
    }
    iCount++;
  }

}
public class SinglyLL_Display 
{
  public static void main(String A[])
  {
    SinglyLL sobj = new SinglyLL();

    sobj.InsertFirst(30);
    sobj.InsertFirst(20);
    sobj.InsertFirst(10);

    sobj.InsertLast(40);
    sobj.InsertLast(50);
    sobj.InsertLast(60);

    sobj.Display();
  }
  
}
